lib/crawler.js

```diff
@@ -1378,21 +1380,21 @@ Crawler.prototype.handleResponse = function(queueItem, response, timeCommenced)
         queueItem.status = "redirected";
 
         // Parse the redirect URL ready for adding to the queue...
-        parsedURL = crawler.processURL(response.headers.location, queueItem);
+        newQueueItem = crawler.processURL(response.headers.location, queueItem);
 
         // Emit redirect event
-        crawler.emit("fetchredirect", queueItem, parsedURL, response);
+        crawler.emit("fetchredirect", queueItem, newQueueItem, response);
 
         if (crawler._isFirstRequest) {
-            parsedURL.depth = 1;
+            newQueueItem.depth = 1;
         }
 
         if (crawler.allowInitialDomainChange && crawler._isFirstRequest) {
-            crawler.host = parsedURL.host;
+            crawler.host = newQueueItem.host;
         }
 
         // Clean URL, add to queue...
-        crawler.queueURL(parsedURL, queueItem);
+        crawler.queueURL(newQueueItem, queueItem);
```

Comment:
Maybe we can rename `queueItem` here to be 'referringQueueItem` or something, to make it clearer.